File: server/go/tests/test_all.go
package main

import (
	"fmt"
	"io/ioutil"
	"os"
	"path/filepath"
	"sort"
	"strings"
)

// Colors for output
const (
	GREEN  = "\033[92m"
	RED    = "\033[91m"
	YELLOW = "\033[93m"
	BLUE   = "\033[94m"
	RESET  = "\033[0m"
)

type checkResult struct {
	ok      bool
	message string
}

func checkSyntax(filePath string) checkResult {
	// Go compiler will check syntax when we try to build
	// For now, just check if file is readable
	_, err := ioutil.ReadFile(filePath)
	if err != nil {
		return checkResult{ok: false, message: fmt.Sprintf("✗ Error reading file: %v", err)}
	}
	return checkResult{ok: true, message: "✓ Syntax valid (file readable)"}
}

func checkImports(filePath string) checkResult {
	content, err := ioutil.ReadFile(filePath)
	if err != nil {
		return checkResult{ok: false, message: fmt.Sprintf("✗ Error reading file: %v", err)}
	}

	contentStr := string(content)
	
	// Check for required imports
	hasGodotenv := strings.Contains(contentStr, "github.com/joho/godotenv") || strings.Contains(contentStr, "godotenv")
	hasNetHttp := strings.Contains(contentStr, "net/http")
	hasJSON := strings.Contains(contentStr, "encoding/json")
	
	missing := []string{}
	if !hasGodotenv {
		missing = append(missing, "github.com/joho/godotenv")
	}
	if !hasNetHttp {
		missing = append(missing, "net/http")
	}
	if !hasJSON {
		missing = append(missing, "encoding/json")
	}
	
	if len(missing) > 0 {
		return checkResult{ok: false, message: fmt.Sprintf("✗ Missing imports: %s", strings.Join(missing, ", "))}
	}
	
	return checkResult{ok: true, message: "✓ Imports valid"}
}

func checkStructure(filePath string) checkResult {
	content, err := ioutil.ReadFile(filePath)
	if err != nil {
		return checkResult{ok: false, message: fmt.Sprintf("✗ Error reading file: %v", err)}
	}

	contentStr := string(content)
	
	checks := []string{}
	if strings.Contains(contentStr, "godotenv.Load") || strings.Contains(contentStr, "godotenv.Overload") {
		checks = append(checks, "Environment loading")
	}
	if strings.Contains(contentStr, "COURIER_API_KEY") {
		checks = append(checks, "API key reference")
	}
	if strings.Contains(contentStr, "http.NewRequest") || strings.Contains(contentStr, "http.Get") || strings.Contains(contentStr, "http.Post") {
		checks = append(checks, "HTTP client")
	}
	if strings.Contains(contentStr, "Authorization") && strings.Contains(contentStr, "Bearer") {
		checks = append(checks, "Authorization header")
	}
	
	if len(checks) >= 2 {
		return checkResult{ok: true, message: fmt.Sprintf("✓ Structure valid (%s)", strings.Join(checks, ", "))}
	} else {
		return checkResult{ok: false, message: "✗ Missing key components"}
	}
}

func main() {
	// Get parent directory (server/go)
	// When run with `go run`, the working directory is the tests directory
	// So we need to go up one level
	testDir, err := os.Getwd()
	if err != nil {
		testDir = "."
	}
	// If we're in the tests directory, go up one level
	if filepath.Base(testDir) == "tests" {
		testDir = filepath.Dir(testDir)
	}
	parentDir := testDir
	
	// Find all .go files in parent directory
	files, err := ioutil.ReadDir(parentDir)
	if err != nil {
		fmt.Fprintf(os.Stderr, "%sError reading directory: %v%s\n", RED, err, RESET)
		os.Exit(1)
	}
	
	var goFiles []string
	for _, file := range files {
		if !file.IsDir() && strings.HasSuffix(file.Name(), ".go") && !strings.HasPrefix(file.Name(), "test_") {
			goFiles = append(goFiles, filepath.Join(parentDir, file.Name()))
		}
	}
	
	sort.Strings(goFiles)
	
	if len(goFiles) == 0 {
		fmt.Printf("%sNo Go files found to test%s\n", RED, RESET)
		os.Exit(1)
	}
	
	fmt.Printf("%sTesting %d Go sample files...%s\n\n", BLUE, len(goFiles), RESET)
	
	type result struct {
		name string
		ok   bool
	}
	results := []result{}
	
	for _, filePath := range goFiles {
		fileName := filepath.Base(filePath)
		fmt.Printf("%sTesting: %s%s\n", BLUE, fileName, RESET)
		
		syntax := checkSyntax(filePath)
		imports := checkImports(filePath)
		structure := checkStructure(filePath)
		
		if syntax.ok {
			fmt.Printf("  %s%s%s\n", GREEN, syntax.message, RESET)
		} else {
			fmt.Printf("  %s%s%s\n", RED, syntax.message, RESET)
		}
		
		if imports.ok {
			fmt.Printf("  %s%s%s\n", GREEN, imports.message, RESET)
		} else {
			fmt.Printf("  %s%s%s\n", YELLOW, imports.message, RESET)
		}
		
		if structure.ok {
			fmt.Printf("  %s%s%s\n", GREEN, structure.message, RESET)
		} else {
			fmt.Printf("  %s%s%s\n", YELLOW, structure.message, RESET)
		}
		
		allOk := syntax.ok && imports.ok && structure.ok
		results = append(results, result{name: fileName, ok: allOk})
		fmt.Println()
	}
	
	// Summary
	fmt.Printf("%s%s%s\n", BLUE, strings.Repeat("=", 60), RESET)
	fmt.Printf("%sSummary:%s\n\n", BLUE, RESET)
	
	passed := 0
	for _, result := range results {
		if result.ok {
			fmt.Printf("  %s✓ PASS%s - %s\n", GREEN, RESET, result.name)
			passed++
		} else {
			fmt.Printf("  %s⚠ PARTIAL%s - %s\n", YELLOW, RESET, result.name)
		}
	}
	
	fmt.Printf("\n%sResults: %d/%d files passed all checks%s\n", BLUE, passed, len(results), RESET)
	
	if passed == len(results) {
		fmt.Printf("%sAll files are valid! ✓%s\n", GREEN, RESET)
		os.Exit(0)
	} else {
		fmt.Printf("%sSome files need attention (likely missing dependencies)%s\n", YELLOW, RESET)
		fmt.Printf("%sRun: go mod tidy && go get%s\n", YELLOW, RESET)
		os.Exit(1)
	}
}

File: server/kotlin/tests/test_all.kt
import java.io.File

// Colors for output
val GREEN = "\u001B[92m"
val RED = "\u001B[91m"
val YELLOW = "\u001B[93m"
val BLUE = "\u001B[94m"
val RESET = "\u001B[0m"

fun checkSyntax(filePath: File): Pair<Boolean, String> {
    try {
        // Use kotlinc to check syntax (compile only, don't run)
        val tempJar = File.createTempFile("test_${filePath.nameWithoutExtension}", ".jar")
        val process = ProcessBuilder("kotlinc", filePath.absolutePath, "-include-runtime", "-d", tempJar.absolutePath)
            .redirectErrorStream(true)
            .start()
        val exitCode = process.waitFor()
        val error = process.inputStream.bufferedReader().readText()
        tempJar.delete()
        
        if (exitCode == 0) {
            return Pair(true, "✓ Syntax valid")
        } else {
            val errorMsg = error.lines().filter { it.contains("error:") }.firstOrNull() ?: error.take(100)
            return Pair(false, "✗ Syntax error: $errorMsg")
        }
    } catch (e: Exception) {
        return Pair(false, "✗ Error checking syntax: ${e.message}")
    }
}

fun checkImports(filePath: File): Pair<Boolean, String> {
    try {
        val content = filePath.readText()
        
        // Check for required patterns
        val hasHttp = content.contains("HttpURLConnection") || content.contains("java.net")
        val hasEnv = content.contains("loadEnv") || content.contains(".env")
        
        if (!hasHttp) {
            return Pair(false, "✗ Missing HTTP client (java.net.HttpURLConnection)")
        }
        if (!hasEnv) {
            return Pair(false, "✗ Missing environment loading")
        }
        
        return Pair(true, "✓ Imports valid")
    } catch (e: Exception) {
        return Pair(false, "✗ Error checking imports: ${e.message}")
    }
}

fun checkStructure(filePath: File): Pair<Boolean, String> {
    try {
        val content = filePath.readText()
        
        val checks = mutableListOf<String>()
        if (content.contains("loadEnv") || content.contains(".env")) {
            checks.add("Environment loading")
        }
        if (content.contains("COURIER_API_KEY")) {
            checks.add("API key reference")
        }
        if (content.contains("HttpURLConnection") || content.contains("URL")) {
            checks.add("HTTP client")
        }
        if (content.contains("Authorization") && content.contains("Bearer")) {
            checks.add("Authorization header")
        }
        
        if (checks.size >= 2) {
            return Pair(true, "✓ Structure valid (${checks.joinToString(", ")})")
        } else {
            return Pair(false, "✗ Missing key components")
        }
    } catch (e: Exception) {
        return Pair(false, "✗ Error checking structure: ${e.message}")
    }
}

fun main() {
    // Get parent directory (server/kotlin)
    val testDir = File(".")
    val parentDir = testDir.parentFile
    
    // Find all .kt files in parent directory
    val allFiles = parentDir.listFiles() ?: arrayOf<File>()
    val ktFiles = allFiles.filter { file ->
        file.isFile && file.name.endsWith(".kt") && !file.name.startsWith("test_")
    }.sortedBy { it.name }
    
    if (ktFiles.isEmpty()) {
        println("${RED}No Kotlin files found to test$RESET")
        kotlin.system.exitProcess(1)
    }
    
    println("${BLUE}Testing ${ktFiles.size} Kotlin sample files...$RESET\n")
    
    val results = mutableListOf<Pair<String, Boolean>>()
    for (file in ktFiles) {
        val fileName = file.name
        println("${BLUE}Testing: $fileName$RESET")
        
        val syntax = checkSyntax(file)
        val imports = checkImports(file)
        val structure = checkStructure(file)
        
        if (syntax.first) {
            println("  ${GREEN}${syntax.second}$RESET")
        } else {
            println("  ${RED}${syntax.second}$RESET")
        }
        
        if (imports.first) {
            println("  ${GREEN}${imports.second}$RESET")
        } else {
            println("  ${YELLOW}${imports.second}$RESET")
        }
        
        if (structure.first) {
            println("  ${GREEN}${structure.second}$RESET")
        } else {
            println("  ${YELLOW}${structure.second}$RESET")
        }
        
        val allOk = syntax.first && imports.first && structure.first
        results.add(Pair(fileName, allOk))
        println()
    }
    
    // Summary
    println("${BLUE}${"=".repeat(60)}$RESET")
    println("${BLUE}Summary:$RESET\n")
    
    var passed = 0
    for ((name, ok) in results) {
        if (ok) {
            println("  ${GREEN}✓ PASS$RESET - $name")
            passed++
        } else {
            println("  ${YELLOW}⚠ PARTIAL$RESET - $name")
        }
    }
    
    val total = results.size
    println("\n${BLUE}Results: $passed/$total files passed all checks$RESET")
    
    if (passed == total) {
        println("${GREEN}All files are valid! ✓$RESET")
        kotlin.system.exitProcess(0)
    } else {
        println("${YELLOW}Some files need attention$RESET")
        kotlin.system.exitProcess(1)
    }
}
